lib/spack/spack/detection/path.py

```diff
@@ -332,21 +326,16 @@ def by_path(
     # TODO: Packages should be able to define both .libraries and .executables in the future
     # TODO: determine_spec_details should get all relevant libraries and executables in one call
     executables_finder, libraries_finder = ExecutablesFinder(), LibrariesFinder()
-
-    executables_path_guess = (
-        spack.util.environment.get_path("PATH") if path_hints is None else path_hints
-    )
-    libraries_path_guess = [] if path_hints is None else path_hints
```

Comment:
I think here we changed behavior, and now we are using `PATH` as a guess for libraries too. Am I reading this right?

Reply:
I think this would only be a change for non Windows. Windows we've always searched the PATH for both binaries and libraries

Reply:
If we don't want this I'm happy to refactor in a way that only libraries on Windows will search the PATH

Reply:
I don't think on linux it makes much sense to search `PATH` directories for libraries. Maybe it would be better to search in things like `LD_LIBRARY_PATH` or similar (obviously, this new search is subject for another PR, not for this one)

Reply:
Ok, I'll prevent the PATH from being searched on unix

Reply:
Addressed